packages/shortest/src/ai/client.ts

```diff
@@ -43,6 +43,7 @@ export class AIClient {
   }> {
     const maxRetries = 3;
     let attempts = 0;
+    const nonRetryableErrors = [500, 401, 403];
```

Comment:
updated logic to maintain an array of error status codes to avoid retrying @rmarescu

Reply:
I think the array can be used directly on line 57, its purpose is self-documenting.